engine/env/spark/src/main/java/org/datacleaner/spark/SparkRunner.java

```diff
@@ -0,0 +1,101 @@
+/**
+ * DataCleaner (community edition)
+ * Copyright (C) 2014 Neopost - Customer Information Management
+ *
+ * This copyrighted material is made available to anyone wishing to use, modify,
+ * copy, or redistribute it subject to the terms and conditions of the GNU
+ * Lesser General Public License, as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this distribution; if not, write to:
+ * Free Software Foundation, Inc.
+ * 51 Franklin Street, Fifth Floor
+ * Boston, MA  02110-1301  USA
+ */
+package org.datacleaner.spark;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+import java.util.UUID;
+
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.spark.launcher.SparkLauncher;
+import org.datacleaner.spark.utils.HadoopUtils;
+
+/**
+ * Prepares for Spark launching from
+ */
+public class SparkRunner {
+    public final static String DATACLEANER_DIR = "/datacleaner";
+    public final static String DATACLEANER_TEMP_DIR = DATACLEANER_DIR + "/temp";
+    public final static String DATACLEANER_LIB_DIR = DATACLEANER_DIR + "/lib";
+
+    private final FileSystem _hadoopDefaultFS;
+    private final ApplicationDriver _applicationDriver;
+    private final URI _configurationFile;
+    private final URI _jobFile;
+    private final URI _resultFile;
+    private final File _hadoopConfigurationDirectory;
+
+    public SparkRunner(String configurationFilePath, String jobFilePath, String resultFilePath) throws IOException {
+        try {
+            _hadoopConfigurationDirectory = HadoopUtils.getHadoopConfigurationDirectoryToUse();
+            _hadoopDefaultFS = FileSystem.newInstance(HadoopUtils.getHadoopConfiguration(_hadoopConfigurationDirectory));
```

Comment:
Seems to me like the HadoopUtils class should have a single method for getting this FileSystem ... At least it seems like intricate code that, if we need to change it, would be difficult to ensure we get it right everywhere.